crates/core/src/operations/transaction/state.rs

```diff
@@ -214,6 +214,14 @@ impl<'a> PruningStatistics for AddContainer<'a> {
         ScalarValue::iter_to_array(values).ok()
     }
 
+    /// return the number of rows for the named column in each container
+    /// as an `Option<UInt64Array>`.
+    ///
+    /// Note: the returned array must contain `num_containers()` rows
+    fn row_counts(&self, column: &Column) -> Option<ArrayRef> {
+        todo!()
```

Comment:
@ion-elgreco I've prepared the arrow version upgrade, but I have no clue on how to implement this new datafusion feature (from https://github.com/apache/arrow-datafusion/pull/9223)

Reply:
@lasantosr it should be similar I guess to how `fn null_counts` is written. On each `add` action you can do `get_stats` on this struct you can fetch the `num_records`